lib/maths/CPoissonMeanConjugate.cc

```diff
@@ -548,17 +547,17 @@ CPoissonMeanConjugate::jointLogMarginalLikelihood(const TDouble1Vec& samples,
             numberSamples += n;
             sampleSum += n * x;
             // Recall n! = Gamma(n + 1).
-            sampleLogFactorialSum += n * boost::math::lgamma(x + 1.0);
+            sampleLogFactorialSum += n * std::lgamma(x + 1.0);
         }
 
         // Get the implied shape parameter for the gamma distribution
         // including the samples.
         double impliedShape = m_Shape + sampleSum;
         double impliedRate = m_Rate + numberSamples;
 
-        result = boost::math::lgamma(impliedShape) + m_Shape * std::log(m_Rate) -
+        result = std::lgamma(impliedShape) + m_Shape * std::log(m_Rate) -
                  impliedShape * std::log(impliedRate) - sampleLogFactorialSum -
-                 boost::math::lgamma(m_Shape);
+                 std::lgamma(m_Shape);
```

Comment:
Check not NaN and is finite and remove redundant try catch.

Reply:
its checked already on line 566

Reply:
Ok cool. But try catch can go here I think.